tests/tests/test_fault_tolerance.py

```diff
@@ -222,9 +222,30 @@ def test_image_download_retry_timeout(
 
         with mender_device.get_reboot_detector(host_ip) as reboot:
             if test_set["blockAfterStart"]:
-                # Block after we start the download.
-                deployment_id, new_yocto_id = common_update_procedure(valid_image)
-                mender_device.run("fuser -mv %s" % (inactive_part))
+                # Use an state script to pause the deployment so that
+                # the test can manipulate the network half way.
+                script_content = """#!/bin/sh
+
+set -e
+touch /tmp/wait-for-install
+while [ -f /tmp/wait-for-install ]; do
+    echo Waiting...
+    sleep 10
+done
+exit 0"""
+                try:
+                    f = open("ArtifactInstall_Enter_00", "w")
+                    f.write(script_content)
+                    f.close()
```

Comment:
A more idiomatic pattern would be:

```python
with open("ArtifactInstall_Enter_00", "w") as f:
    f.write(script_content)
```

Reply:
But how do I then get the `finally` catch to remove the file regardless or errors?

Reply:
Well, you know the name, so you don't need `f.name`, do you? But it's not important, your way is fine too.

Reply:
Ah, sorry I got confused. You just meant adding the `with` context manager but of course keeping the try/finally there. That I can do :)